structuralcodes/sections/_rc_utils.py

```diff
@@ -0,0 +1,84 @@
+    """Calculates the cracked section properties of a reinforced concrete
+    section.  (GenericSection). Materials in surface geometries and point
+    geometries are elastic-linear  in order to make the cracking properties
+    independent of the stress state.  Tension in all surface geometries is
+    neglected.
+
+    Args:
+        section: GenericSection
+        theta: Angle of the neutral axis to the horizontal. theta=0 implies
+               upper compression block.
+
+    Returns:
+        cracked_prop : GrossProperties data of cracked_sec (i.e cracked
+                       section properties)
+        cracked_sec: GenericSection removing tension areas in geometries.
+    """
+
+    def create_surface_geometries(polygons_list, material):
+        """Process shapely polygons to SurfaceGeometries."""
+        # Create an empty list to store SurfaceGeometry objects
+        surface_geometries = []
+
+        # Iterate over the list of polygons and create SurfaceGeometry for each
+        for polygon in polygons_list:
+            # Create a new SurfaceGeometry for the current polygon
+            surface_geometry = SurfaceGeometry(polygon, material)
+            # Add the new SurfaceGeometry to the list
+            surface_geometries.append(surface_geometry)
+
+        return surface_geometries
+
+    if not section.geometry.reinforced_concrete:
+        return None
+
+    rotated_geometry = section.geometry.rotate(theta)
+
+    for geo in rotated_geometry.geometries:
+        Ec = geo.material.get_tangent(eps=0)
+        elastic_concrete = UserDefined([-100, 0], [-100 * Ec, 0])
+        geo._material = elastic_concrete
+
+    for pg in rotated_geometry.point_geometries:
+        Es = pg.material.get_tangent(eps=0)
+        elastic_steel = Elastic(Es, 'elastic steel')
+        pg._material = elastic_steel
+
+    curv = -1e-5  # Any curvature should return the same mechanical properties.
+    # Find the equilibrium with fixed curvature
+    eps = section.section_calculator.find_equilibrium_fixed_curvature(
+        rotated_geometry, 0, curv, 0
+    )[0]
+    z_na = -eps / curv  # distance to neutral fibre
+
+    # Cutting concrete geometries and retaining the compressed block
+    cut_geom = CompoundGeometry(None, None)
+    for i, part in enumerate(rotated_geometry.geometries):
+        min_x, max_x, min_y, max_y = part.calculate_extents()
+        upper_div, lower_div = part.split(((min_x, z_na), 0))
+        # Convert to SurfaceGeometry
+        subpart_sg = create_surface_geometries(upper_div, part.material)
+        if i == 0:
+            cut_geom = CompoundGeometry(subpart_sg)
+        else:
+            cut_geom += subpart_sg
+
+    # Add reinforcement geometries
+    for reinf in rotated_geometry.point_geometries:
+        cut_geom += reinf
+
+    # return geoemtry to original rotation
+    cracked_geom = cut_geom.rotate(-theta)
+
+    # Define the cracked section
+    cracked_sec = GenericSection(cracked_geom)
```

Comment:
in this way we are not retaining the user desired integration method I think...
1) we should take the integrator and the mesh size from the original section. Furthermore the mesh_size being a percentage of the area, should be scaled accordingly.
2) is it better to return the compound geometry or the section?

Reply:
Perhaps it would be better to return cracked_geom (CompoundGeometry) directly and not the section. I'm not sure if returning the complete GenericSection object adds any value. What do you think?

Reply:
This makes perfect sense, @MestreCarlos. As I see it, the reason for returning either the section or the geometry is for letting the user check what was the basis for the calculations, and the geometry is the exactly the basis for this.

However, when the `cracked_sec` is initialized, the integrator, and mesh size if any, should be inherited from the section that was input to the function in order to maintain user control, as commented by @talledodiego above.